### PR TITLE
feat(datasets): Added the Experimental pypdf.PDFDataset

### PR DESCRIPTION
**Pull request by jitu5**
# Description
Resolve https://github.com/kedro-org/kedro-plugins/issues/1180 

This PR adds a new experimental dataset `pypdf.PDFDataset` to handle PDF file reading and writing, with support for password-protected PDFs and various PDF operations.

## Development notes
**Adds pypdf.PDFDataset** - A new experimental dataset for loading PDF files using the pypdf library

**Complete implementation following established patterns:**
- Full load functionality with pypdf.PdfReader objects
- Support for password-protected PDFs with encryption
- Multiple filesystem support (local, S3, GCS, etc.)
- Comprehensive error handling
- Cache invalidation
- Flexible load arguments (strict mode, password support)

**Full test coverage** - Unit tests covering all functionality including edge cases, password protection, and multi-page PDFs

**Documentation** - Complete API documentation and integration with mkdocs

**Proper dependency management** - New optional dependencies: `pypdf-pdfdataset = ["pypdf>=3.0.0", "cryptography>=3.1"]`

## Usage

```yaml
# YAML API

# Basic PDF loading
my_pdf:
  type: pypdf.PDFDataset
  filepath: data/document.pdf

# Password-protected PDF
secure_pdf:
  type: pypdf.PDFDataset
  filepath: data/secure.pdf
  load_args:
    password: "secret123"
    strict: false
```

```python
# Python API
from pypdf import PdfReader, PdfWriter
from kedro_datasets_experimental.pypdf import PDFDataset

# Load a PDF
dataset = PDFDataset(filepath="document.pdf")
pdf_reader = dataset.load()
print(f"Number of pages: {len(pdf_reader.pages)}")

# Load password-protected PDF
secure_dataset = PDFDataset(
    filepath="secure.pdf",
    load_args={"password": "secret123", "strict": False}
)
pdf_reader = secure_dataset.load()
```

## Checklist

- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [x] Updated the documentation to reflect the code changes
- [ ] Updated `jsonschema/kedro-catalog-X.XX.json` if necessary
- [x] Added a description of this change in the relevant `RELEASE.md` file
- [x] Added tests to cover my changes
- [ ] Received approvals from at least half of the TSC (required for adding a new, non-experimental dataset)

### CODE REVIEW

File: kedro-datasets/kedro_datasets_experimental/pypdf/pdf_dataset.py [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+from kedro.io.core import (
+    AbstractVersionedDataset,
+    DatasetError,
+    Version,
+    get_filepath_str,
+    get_protocol_and_path,
+)
+
+
+class PDFDataset(AbstractVersionedDataset[NoReturn, list[str]]):
```
Comment (ankatiyar) [inline kedro-datasets/kedro_datasets_experimental/pypdf/pdf_dataset.py]: I think this dataset can be `AbstractDataset` instead of `AbstractVersionedDataset` since it's a read-only
